src/jit/valuenum.cpp

```diff
@@ -245,6 +245,35 @@ TFp FpRem(TFp dividend, TFp divisor)
     return (TFp)fmod((double)dividend, (double)divisor);
 }
 
+template <typename TFp, typename TFpTraits>
+TFp FpDiv(TFp dividend, TFp divisor)
+{
+#ifdef _TARGET_ARM_
+    // The following is needed since NaN returned by floating point division on ARM32
+    // has different binary represenation (0x7fc00000 ~ float, 0x7ff8000000000000 ~ double)
+    // from NaN on X86 and X64 (0xffc00000 ~ float, 0xfff8000000000000 ~ double).
+    // As a consequence, this causes X86_ARM32 and ARM32 crossgens produce different binary output.
+
+    // From the ECMA standard:
+    //
+    // If [dividend] is zero and [divisor] is zero
+    //   the result is NaN.
+    // If [dividend] is infinity and [divisor] is infinity
+    //   the result is NaN.
+
+    if (dividend == 0 && divisor == 0)
+    {
+        return TFpTraits::NaN();
```

Comment:
My preference would be to also conditionalize the Traits classes to behave like the target (e.g. use the ARM default-NaN value for ARM).  Otherwise you get the possibility of AoT'd code disagreeing with JIT'd results.  Possibly contrived, but it's a pretty easy change to make.